build.zig

```diff
@@ -1273,9 +1274,44 @@ fn build_rust_client(
     options: struct {
         vsr_module: *std.Build.Module,
         vsr_options: *std.Build.Step.Options,
+        tb_client_header: *Generated,
         mode: std.builtin.OptimizeMode,
     },
 ) void {
+    step_clients_rust.dependOn(&options.tb_client_header.step);
+
+    inline for (platforms) |platform| {
+        const query = Query.parse(.{
+            .arch_os_abi = platform[0],
+            .cpu_features = platform[2],
+        }) catch unreachable;
+        const resolved_target = b.resolveTargetQuery(query);
+
+        const root_module = b.createModule(.{
+            .root_source_file = b.path("src/tigerbeetle/libtb_client.zig"),
+            .target = resolved_target,
+            .optimize = options.mode,
+        });
+        root_module.addImport("vsr", options.vsr_module);
+        root_module.addOptions("vsr_options", options.vsr_options);
+        if (options.mode == .ReleaseSafe) strip_root_module(root_module);
+
+        const static_lib = b.addLibrary(.{
+            .name = "tb_client",
+            .linkage = .static,
+            .root_module = root_module,
+        });
+        static_lib.bundle_compiler_rt = true;
+        static_lib.pie = true;
+        static_lib.linkLibC();
```

Comment:
A reminder for myself: the reason we link libc is because we are spawning a thread inside tb_client, and threads are part of C runtime, due to thread locals.